hxmath/math/Vector2.hx

```diff
@@ -839,5 +839,20 @@ abstract Vector2(Vector2Type) from Vector2Type to Vector2Type
         return self.clone()
             .rotateRight();
     }
+
+    private inline function set_length(v:Float):Float
+    {
+        var self:Vector2 = this;
+        self.normalizeTo(v);
+        return v;
+    }
+
+    private inline function set_angle(v:Float):Float
+    {
+        var self:Vector2 = this;
```

Comment:
Can you add a test case for this? Check that you get consistent values when setting/getting the angle. The math itself looks right to me.

Reply:
Sure thing 👍

Reply:
Test case added to `Test2D`, hopefully that's the right spot!

Reply:
Can you add a non-random test? Maybe something that tests 8 directions in a for-loop. Random is probably overkill for something like angle.

Reply:
Done. I do want to point out that in the new test, I was getting negative values from `Vector2.angle` (I believe that's just something that happens when you use Math.atan2 to get a Vector2's angle), so I had to wrap the output angle before checking for equality in order for the test to pass. Here's a sample of the angle inputs/outputs so you can see what I mean:

```
./test/Test2D.hx:183: input angle 0, output angle: 0
./test/Test2D.hx:183: input angle 0.785398163397448, output angle: 0.785398163397448
./test/Test2D.hx:183: input angle 1.5707963267949, output angle: 1.5707963267949
./test/Test2D.hx:183: input angle 2.35619449019234, output angle: 2.35619449019234
./test/Test2D.hx:183: input angle 3.14159265358979, output angle: 3.14159265358979
./test/Test2D.hx:183: input angle 3.92699081698724, output angle: -2.35619449019235
./test/Test2D.hx:183: input angle 4.71238898038469, output angle: -1.5707963267949
./test/Test2D.hx:183: input angle 5.49778714378214, output angle: -0.785398163397448
```

Not sure if this is OK though, so I thought id mention it.

Reply:
Yeah, that makes sense. atan2 gives you angles in (-pi, pi].